lib/ui/natives.dart

```diff
@@ -4,10 +4,27 @@
 
 // TODO(dnfield): remove unused_element ignores when https://github.com/dart-lang/sdk/issues/35164 is resolved.
 
-
 // @dart = 2.12
 part of dart.ui;
 
+// ignore_for_file: avoid_classes_with_only_static_members
+
+/// Helper functions for Dart Plugin Registrants.
+class DartPluginRegistrant {
+  static bool _wasInitialized = false;
+
+  /// Makes sure the that the Dart Plugin Registrant has been called for this
+  /// isolate. This can safely be executed multiple times on the same isolate,
+  /// but should not be called on the Root isolate.
```

Comment:
Is the app developer responsible for calling this method?

Reply:
Yep, potentially plugin developers too.

Reply:
if the check `_wasInitialized ` is moved to `DartPluginRegistrant_EnsureInitialized`, would it handle the root isolate case?

Reply:
Nope, that wouldn't allow it to be called from the background isolates which is what we want.  If we did that we'd have to keep a map between the isolate and a bool to determine if it was called or not.  That would be a bit involved since it would mean uniquely identifying the isolate some how and exposing that to c++.

Reply:
C++ side could only check for the root isolate, and the dart side for the custom isolate. Maybe the map is cleaner.

Reply:
If we think that this is a problem, the easiest solution would be to transfer the bool into the _PluginRegistrant and making the registrant idempotent.

Reply:
sgtm

Reply:
I filed an issue for it: https://github.com/flutter/flutter/issues/99444